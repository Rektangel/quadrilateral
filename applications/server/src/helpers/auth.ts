import jwt from 'jsonwebtoken';
import { User } from '../models/User';

const { JWT_SECRET, JWT_EXPIRE_TIME } = process.env;

interface JwtPayload {
  id: string;
}

export const generateJwtToken = (user: User): string => {
  const payload: JwtPayload = {
    id: user.id,
  };

  return jwt.sign(payload, JWT_SECRET, { expiresIn: JWT_EXPIRE_TIME });
};
