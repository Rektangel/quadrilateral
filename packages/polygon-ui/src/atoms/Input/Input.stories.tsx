import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, radios, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Input from './Input';

storiesOf('Input', module).add(
  'Default',
  (): JSX.Element => (
    <Input
      type={radios('Type', { Text: 'text', Password: 'password' }, 'text')}
      name="input"
      onChange={action('Input changed')}
      placeholder={text('Placeholder', '')}
      disabled={boolean('Disabled', false)}
      error={boolean('Error', false)}
    />
  )
);
