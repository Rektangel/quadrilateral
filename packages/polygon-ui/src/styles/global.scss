@import './abstracts/colors';

[data-theme='light'] {
  --global-background-color: #{$color-gray-100};
  --global-color: #{$color-blue-gray-800};
}

[data-theme='dark'] {
  --global-background-color: #{$color-blue-gray-800};
  --global-color: #{$color-blue-gray-100};
}

body {
  background-color: var(--global-background-color);
  color: var(--global-color);
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}
