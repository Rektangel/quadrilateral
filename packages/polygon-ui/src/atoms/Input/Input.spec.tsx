import * as React from 'react';
import { renderWithTheme } from '../../helpers/testHelper';
import { dark, light } from '../../styles/themes';
import Input from './Input';

describe('Input', (): void => {
  it('with light theme renders with no errors', (): void => {
    const { asFragment } = renderWithTheme(
      <Input name="input-light" type="text " />,
      light
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('with dark theme renders with no errors', (): void => {
    const { asFragment } = renderWithTheme(
      <Input name="input-dark" type="text " />,
      dark
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
