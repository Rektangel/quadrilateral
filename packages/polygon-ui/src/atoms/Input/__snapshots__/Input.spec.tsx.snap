// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input with dark theme renders with no errors 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnylx gfnHGu"
    name="input-dark"
    type="text "
  />
</DocumentFragment>
`;

exports[`Input with light theme renders with no errors 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnylx cBWbsb"
    name="input-light"
    type="text "
  />
</DocumentFragment>
`;
